### PR TITLE
Add a buildExportAllLoose template

### LINKED ISSUE #4097: export-import uses Object.defineProperty even when es6.modules is in loose mode (T7116)
> Issue originally made by @mnpenner
### Bug information
- **Babel version:** 6.5.2
- **Node version:** v4.2.3
- **npm version:** 2.14.15
### Options

Snippet from my `webpack.config.js`:

```
var babelPlugins = [
    // 'add-module-exports',
    'syntax-flow',
    'transform-flow-strip-types',
    'transform-es3-member-expression-literals', // http://stackoverflow.com/a/33513000/65387
    'transform-es3-property-literals',
    'transform-es5-property-mutators',
    ['transform-es2015-modules-commonjs', {loose: true}], // https://babeljs.slack.com/archives/discussion/p1455667909000424
];

{
    test: /\.js$/,
    include: assetsDir,
    loader: 'babel',
    query: {
        presets: ['es2015-webpack', 'stage-1'],
        plugins: babelPlugins,
        compact: false,
    }
},
```
### Input code

```
export { default as notify } from './wxu.notify';
```
### Description

This:

```
export { default as notify } from './wxu.notify';
```

Roughly translates into this:

```
var _wxu, _;
exports.__esModule = true;
_wxu = __webpack_require__(431);
Object.defineProperty(exports, "notify", {
    enumerable: true,
    get: function() {
        return _interopRequireDefault(_wxu)["default"];
    }
});
```

It should avoid `Object.defineProperty` as [stated here](https://babeljs.algolia.com/docs/advanced/loose/#es6-modules).

[Discussion here](https://babeljs.slack.com/archives/discussion/p1455669116000443)

### PR DESCRIPTION
**Pull request by matiasfh**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues             | Fixes #4097 
| Patch: Bug Fix?          |  n
| Major: Breaking Change?  |  n
| Minor: New Feature?      |  n
| Tests Added/Pass?        |  n/y
| Spec Compliancy?         |  n
| License                  | MIT
| Doc PR                   | 
| Any Dependency Changes?  |  

Add a new template for `buildExportAll` on `es2015-modules-common` to be used for `loose` mode as described in #4097 

Looks like this shouldn't be merged until #6230 get's merged

### CODE REVIEW

**Comment by existentialism** [comment]
@msdark thanks for the PR! can you add a test as well?

**Comment by msdark** [comment]
Sure, this is my first PR .. will add the test during the day. (just checking how the test suite works)

File: packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js [hunk 1]
```diff
@@ -460,12 +467,17 @@ export default function() {
...
-              const exportNode = buildExportAll({
+              const obj = {
                 OBJECT: addRequire(
                   path.node.source.value,
                   path.node._blockHoist,
                 ),
-              });
+              };
+
+              let exportNode = buildExportAll(obj);
```
Comment (existentialism) [inline packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js]: Nit, can avoid calling if `this.opts.loose` `buildExportAll` via ternary?

Comment (msdark) [reply packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js]: `const exportNode = (this.opts.loose)? buildExportAllLoose(obj): buildExportAll(obj);`

**Comment by existentialism** [comment]
@msdark sounds good! be sure to check out the [contributing docs](https://github.com/babel/babel/blob/master/CONTRIBUTING.md) if you haven't already, and feel free to join Slack if you have any questions!

**Comment by msdark** [comment]
Tests added

**Comment by loganfsmyth** [comment]
I landed https://github.com/babel/babel/pull/6230 so this will need to be reworked now.

**Comment by loganfsmyth** [comment]
I think your merge may have gone poorly. The commits in here don't appear to do anything now except revert changes from my PR.

**Comment by msdark** [comment]
I think this can be close... After merge I ran tests and looks like you're pr already include the required changes for this issue so there is nothing to do....Will review my branch asap

Matías Hernández Arellano
Software Engineer
+569161929521

**Comment by loganfsmyth** [comment]
@msdark I haven't added anything new around this in my PR. From what I can see, when you did the git merge, you removed all of the changes you made, so your code is just gone from this PR.

**Comment by existentialism** [comment]
@matiasfh interested in picking this back up?

**Pull request closed without merging**